tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -45,9 +55,41 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
-        var bytes = headers.TryGetLastBytes(DataStreamsPropagationHeaders.PropagationKey);
+        // Try to extract from the base64 header first
+        var base64Bytes = headers.TryGetLastBytes(DataStreamsPropagationHeaders.PropagationKeyBase64);
+        if (base64Bytes is { Length: > 0 })
+        {
+            try
+            {
+                var base64String = Encoding.UTF8.GetString(base64Bytes);
+                var decodedBytes = Convert.FromBase64String(base64String);
```

Comment:
Similar to my comment above, but with `Base64.DecodeFromUtf8` ([docs](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.text.base64.decodefromutf8)), you can skip the intermedia string.

Reply:
Thanks, moved this one over to the new direct byte conversion method as well to avoid intermediate string allocation.